dpath/options.py

```diff
@@ -1 +1,7 @@
 ALLOW_EMPTY_STRING_KEYS = False
+
+#  Extension to interpret path segments "{rrr}" as re.regexp "rrr" enabled by default.
+#  Disable to preserve backwards compatibility in the case where a user has a
+#  path "a/b/{cd}" where the brackets are intentional and do not denote a request
+#  to re.compile cd
+DPATH_ACCEPT_RE_REGEXP = True
```

Comment:
Should this feature be enabled by default? Maybe keep disabled for now and enable in the future? Would love to hear your thoughts.

Reply:
1.  My thinking is that setting the default is the maintainer decision. 
     -  The conservative way is to disable by default, enabling by default makes it easier  for adopters.
     - I guess dpath uses the deprecation mechanism, so that starting conservatively seems right.
3. If I was to do it from scratch now, I would provide a function `do_regexp_in_str` that would recognize the strings that represent regexp, re.compile if required. 
   - Could be initialized to identity (backward compatibility ), or to recognition of round brackets ( and re.compile). 
   - The advantage is that this scheme lets the users redefine the function (this is Python), so that they can 
   use whatever string delimiters they please.

Anyway I now make the disabling the default in the code and the documentation.